src/core/thread/mesh_forwarder.cpp

```diff
@@ -368,9 +368,10 @@ Message *MeshForwarder::GetIndirectTransmission(const Child &aChild)
         mAddMeshHeader = false;
         GetMacSourceAddress(ip6Header.GetSource(), mMacSource);
 
-        if (ip6Header.GetDestination().IsLinkLocal() || ip6Header.GetDestination().IsMulticast())
+        if (ip6Header.GetDestination().IsLinkLocal() && !ip6Header.GetDestination().IsLinkLocalMulticast())
```

Comment:
I don't think the `IsLinkLocalMulticast()` check is necessary.  `IsLinkLocal()` will always return false for multicast addresses.